app/controllers/cases/letters_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Cases::LettersController < ApplicationController
+  before_action :set_case
+
+  def new
+    @type = type_name(params[:type])
+    @letter = Letter.new(params[:letter_template_id])
+    @letter_templates = LetterTemplate.where(template_type: @type)
+  end
+
+  def render_letter
+    @type = params[:type]
+    unless params[:letter_form] and params[:letter_form][:letter_template_id]
```

Comment:
Yes, this `and` is used for flow control and short-circuit exits early, never evaluating the right hand side if the left is false - here it prevents the second part from blowing up if params[:letter_form] is nil

Reply:
The right hand side will not be executed if `params[:letter_form]` is `true` when using `&&`
You can probably replace
```
unless params[:letter_form] and params[:letter_form][:letter_template_id]
```
with
```
unless params.dig(:letter_form, :letter_template_id)
 ...
end
```

Reply:
oh nice